Base/QTCore/qSlicerUtils.cxx

```diff
@@ -51,6 +51,18 @@ bool qSlicerUtils::isExecutableName(const QString& name)
 //------------------------------------------------------------------------------
 bool qSlicerUtils::isCLIExecutable(const QString& filePath)
 {
+  // check if .py file starts with `#!` magic
+  //   note: uses QTextStream to avoid issues with BOM.
+  QFile scriptFile(filePath);
```

Comment:
I suggest to move this code to https://github.com/Slicer/Slicer/blob/master/Base/QTCore/qSlicerUtils.h

Reply:
May be a function named `isCLIScriptedExecutable(cons QString& filePath)`